frontend/src/components/pages/AdminPage.tsx

```diff
@@ -70,27 +71,40 @@ const AdminPage = (): React.ReactElement => {
     setSortOrder,
   };
 
-  const { loading, error, data } = useQuery(GET_USERS_BY_ROLE, {
+  const {
+    loading: adminLoading,
+    error: adminError,
+    data: adminData,
+  } = useQuery(GET_USERS_BY_ROLE, {
     fetchPolicy: "cache-and-network",
     variables: { role: "Admin" },
   });
 
-  const filteredAdmins = React.useMemo(() => {
-    let filteredUsers = data?.usersByRole;
+  const {
+    loading: teacherLoading,
+    error: teacherError,
+    data: teacherData,
+  } = useQuery(GET_USERS_BY_ROLE, {
+    fetchPolicy: "cache-and-network",
+    variables: { role: "Teacher" },
+  });
```

Comment:
Adding this slowed the loading time. Would it be better to have a single query getting both teacher and admin roles, and then filtering in the frontend based on which table they're on? Idk if it's worth it though because we might need to change how the backend works.

Reply:
we have a resolver called `users` that you can use to get all users - you will need to define a new query in `APIClients/queries/UserQueries.ts` that returns the needed fields + role